crates/core/tedge_api/src/entity_store.rs

```diff
@@ -602,59 +602,36 @@ impl EntityStore {
         &mut self,
         entity_topic_id: &EntityTopicId,
     ) -> Result<Vec<EntityRegistrationMessage>, entity_store::Error> {
-        if entity_topic_id.matches_default_topic_scheme() {
-            if entity_topic_id.is_default_main_device() {
-                return Ok(vec![]); // Do nothing as the main device is always pre-registered
-            }
-
-            let mut register_messages = vec![];
-
-            let parent_device_id = entity_topic_id
-                .default_source_device_identifier()
-                .expect("device id must be present as the topic id follows the default scheme");
-
-            if !parent_device_id.is_default_main_device() && self.get(&parent_device_id).is_none() {
```

Comment:
Oops, I removed this check that the device should not already be registered.

```
self.get(&parent_device_id).is_none()
```

This explains why the system tests are failing.

For instance for []():

Before, only the service where registered:

```
c8y/s/us/foo_child1   102,foo:device:foo_child1:service:app1,service,app1,up
```

But this check missing service is also registered, while already known by c8y:

```
c8y/s/us                       101,foo:device:foo_child1,foo_child1,thin-edge.io-child
c8y/s/us/foo_child1   102,foo:device:foo_child1:service:app1,service,app1,up
```

Reply:
Fixed by https://github.com/thin-edge/thin-edge.io/pull/3139/commits/eaf5a8f765f9e6c64631f74b39835ba17a74698a (with a following simplification: https://github.com/thin-edge/thin-edge.io/pull/3139/commits/1d3f7070af0c8faeb7f339c59e9ec73ffda0c1ea)